activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -156,7 +156,13 @@ def sole
     # record is found. Raises ActiveRecord::SoleRecordExceeded if more than one
     # record is found.
     #
-    #   Product.find_sole_by(["price = %?", price])
+    #   Product.find_sole_by(["price = ?", price])
+    #
+    # ==== Edge Cases
+    #
+    #   Product.find_sole_by(nil)         # returns ActiveRecord object if the argument is nil.
```

Comment:
```suggestion
    #   Product.find_sole_by(nil)         # returns the first product if only one product exists on the table or raises ActiveRecord::SoleRecordExceeded if more than one product exits.
```

But this is pretty much the documentation of the method, so not sure if it is an edge case.

Reply:
I did not want to repeat the existing documentation that already specifies the behavior for multiple records. Let me know if I can document the nil case. I think nil cases are useful for beginners.

Reply:
But this is the `nil` case, isn't? It is documenting the behavior for multiple records `Product.find_sole_by(nil)` when there are 2 or more records on that table will raise an error. There is no other possible documentation. How you wrote it is wrong.

Reply:
## Nil Case

- 0 record, it raises ActiveRecord::RecordNotFound: Couldn't find Article
- 1 record   - It returns the record. This is unexpected behavior. Because the nil seems to be ignored and it just returns the record. The returned record is not really a match for nil that was passed in.

For documenting edge cases, can we consider the nil input with 0 record in the database?

Reply:
You are missing the case where you have many record. It will raise `ActiveRecord::SoleRecordExceeded`.

For 1 record the behavior is correct. `Product.find_sole_by(nil)` is the same as `Product.where(nil).sole` which runs `SELECT * FROM products` and should return all the products.

> For documenting edge cases, can we consider the nil input with 0 record in the database?

No. If `nil` input is an edge case, we should document the behavior with any number of records in the database. If we only consider 0, people will expect that passing `nil` always raises `ActiveRecord::RecordNotFound`

Reply:
I can document each case separately. 

Currently I am stuck in terms of providing documentation PRs due to misunderstanding of the term edge case. I did some research to understand edge cases and boundary cases. According to my discussion notes:

Edge cases can consist of very high, very low (these are within accepted input range). It can also consist of maximum and minimum (limit of system’s capacity for 32 bit integer, it will 2 to the power of 31 - 1).

Boundary case is a subset of edge case. The input range limit can be just outside, inside and outside of the boundary values. Example: 0, 1, n, n+1 for input range 1 to n.

Corner case combines multiple edge conditions.

Is my understanding correct? This will unblock me and continue contributing to Rails documentation. So if this is correct, these will go under edge cases heading.

Reply:
That isn't what our guide means with Edge cases. The goal there is to show cases where the behavior of the method will differ from what is already documented as the common case. In this PR, none of the cases you are documenting are different form the common case. In another PR that I merged we had a few.